### PR TITLE
feat: add test

### LINKED ISSUE #8: Add a handled test form and add a test example
### Explanation:

"Test" is something that is required in all projects, for example. If we have some test is easier to change an existing component without having to check if we are breaking something.

Tests are always tedious to add and configure in the project, I think this can be improved if we have a template that already has some default test example to guide us to add tests more easily. And there should be a way to add test in the template.

### Checklist

- [ ]  There is a way to add Test in the template.
- [ ] There is a basic default test in the file to guide us to add the test.

### PR DESCRIPTION
**Pull request by denisosuna**
issue: #8 

Was added support for test

image

### CODE REVIEW

**Review by Israel-Laguan — APPROVED** [review]
@Katsu08 @angelik0828 please check if there is regression

**Review by heliomar-pena — COMMENTED** [review]
Good job! What do you think about adding a simple functionality test (it was called "regression test"?), for example, a test that confirm that if you do click in the button then the counter is sum +1?

**Comment by heliomar-pena** [comment]
Please remove package lock json and I think the PR can be merged

**Comment by Israel-Laguan** [comment]
> Please remove package lock json and I think the PR can be merged

I think is more a rebase in this case. As this PR includes new packages, the lockfile changed.

**Comment by denisosuna** [comment]
@Israel-Laguan @Katsu08 i think is ready to merge

File: .eslintrc.js [hunk 1]
```diff
@@ -41,3 +41,5 @@ module.exports = {
     },
   },
 }
+
```
Comment (Israel-Laguan) [inline .eslintrc.js:44]:
```suggestion
```

File: package.json [hunk 2]
```diff
@@ -29,9 +29,15 @@
   },
   "devDependencies": {
     "@arkweid/lefthook": "~0.7.7",
+    "@babel/preset-env": "^7.16.11",
+    "@babel/preset-react": "^7.16.7",
+    "@babel/preset-typescript": "^7.16.7",
```
Comment (Israel-Laguan) [inline package.json]:
```suggestion
    "@babel/preset-env": "~7.16.11",
    "@babel/preset-react": "~7.16.7",
    "@babel/preset-typescript": "~7.16.7",
```

File: package.json [hunk 3]
```diff
@@ -29,9 +29,15 @@
...
     "@commitlint/cli": "~16.2.1",
     "@commitlint/config-conventional": "~16.2.1",
     "@emotion/react": "~11.8.1",
+    "@testing-library/jest-dom": "^5.16.2",
+    "@testing-library/react": "^12.1.3",
+    "@testing-library/user-event": "^13.5.0",
```
Comment (Israel-Laguan) [inline package.json]:
```suggestion
    "@testing-library/jest-dom": "~5.16.2",
    "@testing-library/react": "~12.1.3",
    "@testing-library/user-event": "~13.5.0",
```

File: package.json [hunk 4]
```diff
@@ -29,9 +29,15 @@
...
     "@arkweid/lefthook": "~0.7.7",
+    "@babel/preset-env": "~7.16.11",
+    "@babel/preset-react": "~7.16.7",
+    "@babel/preset-typescript": "~7.16.7",
     "@commitlint/cli": "~16.2.1",
     "@commitlint/config-conventional": "~16.2.1",
     "@emotion/react": "~11.8.1",
+    "@testing-library/jest-dom": "^5.16.2",
+    "@testing-library/react": "^12.1.3",
+    "@testing-library/user-event": "^13.5.0",
```
Comment (Israel-Laguan) [inline package.json]:
```suggestion
    "@testing-library/jest-dom": "~5.16.2",
    "@testing-library/react": "~12.1.3",
    "@testing-library/user-event": "~13.5.0",
```

File: package.json [hunk 5]
```diff
@@ -49,6 +55,7 @@
     "eslint-plugin-promise": "^5.2.0",
     "eslint-plugin-react": "^7.28.0",
     "eslint-plugin-react-hooks": "^4.3.0",
+    "identity-obj-proxy": "^3.0.0",
```
Comment (Israel-Laguan) [inline package.json]:
```suggestion
    "identity-obj-proxy": "~3.0.0",
```

**Review by Israel-Laguan — APPROVED** [review]

File: src/pages/Home.test.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import { render } from '@testing-library/react'
+import Home from './Home'
+
+describe('App', () => {
+  it('Home component renders correctly', () => {
+    const tree = render(<Home />)
+    expect(tree).toMatchSnapshot()
+  })
+
+  it('Verify if data test exist', () => {
```
Comment (Israel-Laguan) [inline src/pages/Home.test.js]:
```suggestion
  it('Verify if data test id exist on snapshot', () => {
```

File: package.json [hunk 7]
```diff
@@ -49,6 +55,7 @@
     "eslint-plugin-promise": "^5.2.0",
     "eslint-plugin-react": "^7.28.0",
     "eslint-plugin-react-hooks": "^4.3.0",
+    "identity-obj-proxy": "~3.0.0",
```
Comment (Israel-Laguan) [inline package.json]:
```suggestion
    "eslint-plugin-promise": "~5.2.0",
    "eslint-plugin-react": "~7.28.0",
    "eslint-plugin-react-hooks": "~4.3.0",
    "identity-obj-proxy": "~3.0.0",
```

**Pull request merged**